core/store/src/trie/mod.rs

```diff
@@ -895,7 +895,7 @@ mod tests {
         assert_eq!(pairs, iter_pairs);
 
         let mut other_iter = trie.iter().unwrap();
-        other_iter.seek(b"r").unwrap();
+        other_iter.seek_prefix(b"r").unwrap();
         assert_eq!(other_iter.next().unwrap().unwrap().0, b"x".to_vec());
```

Comment:
```suggestion
        assert_eq!(None, other_iter.next().unwrap());
```

And how about also adding a case which seeks to `b"x"` and test verifies that only `b"x"` entry is returned.

Reply:
>         assert_eq!(None, other_iter.next().unwrap());
Doesn't compile. 

> And how about also adding a case which seeks to b"x" and test verifies that only b"x" entry is returned.
I don't understand what you want to test here.

Reply:
I guess `assert_eq!(None, other_iter.next());`.  Maybe it’s worth adding a helper functions here.  Something like:

```
let assert_has_next = |want| {
    assert_eq!(Some(want), other_iter.next().map(|item| item.unwrap().0).as_deref());
};
```

I’m not 100% sure if it’ll compile.

That when seeking to ‘x’ only ‘x’ is returned.  Since now seeking to ‘r’ doesn’t return anything (since ‘r’ is not a prefix of any of the keys).

Reply:
updated